src/main/java/io/beanmapper/core/converter/impl/RecordToAnyConverter.java

```diff
@@ -46,11 +47,15 @@ public boolean match(Class<?> sourceClass, Class<?> targetClass) {
      * @param <T> The type of the intermediary class.
      */
     private <R extends Record, T> T copyRecordComponentsToIntermediary(R recordClass, T intermediary)
-            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
+            throws NoSuchFieldException, IllegalAccessException, NoSuchMethodException {
         Map<String, Object> fieldMap = new HashMap<>();
         for (var field : intermediary.getClass().getDeclaredFields()) {
-            Method getter = recordClass.getClass().getDeclaredMethod(field.getName());
-            fieldMap.put(field.getName(), getter.invoke(recordClass));
+            MethodHandle methodHandle = MethodHandles.lookup().unreflect(recordClass.getClass().getDeclaredMethod(field.getName()));
+            try {
+                fieldMap.put(field.getName(), methodHandle.invoke(recordClass));
+            } catch (Throwable t) {
```

Comment:
Meestal geen goed idee om Throwable te catchen. Waarom staan er dan exceptions in de throws clause van de method?

Reply:
MethodHandle.invoke throws helaas ook expliciet Throwable.